app/models/motd_file.rb

```diff
@@ -1,67 +1,43 @@
+require 'open-uri'
+
 class MotdFile
-  attr_reader :motd_system_file
 
-  Message = Struct.new :date, :title, :body do
-    def self.from(str)
-      if str =~ /(\d+[-\/\.]\d+[-\/\.]\d+)\n--- ([ \S ]*)\n(.*)/m
-        MotdFile::Message.new(Date.parse($1), $2, $3)
-      else
-        nil
-      end
-    rescue ArgumentError => e
-      Rails.logger.warn("MOTD message poorly formatted: #{e} => #{e.message}")
+  attr_reader :motd_uri
 
-      nil
-    end
-  end
   # Initialize the Motd Controller object based on the current user.
   #
-  # @param [boolean] update_user_view_timestamp True to update the last viewed timestamp. (Default: false)
-  def initialize(path = ENV['MOTD_PATH'], update_user_view_timestamp: false)
-    @motd_system_file = path
-
-    touch if update_user_view_timestamp
-  end
-
-  # An empty file whose modification timestamp indicates the last time the user
-  # viewed the motd messages. This is useful for when we want to use the file
-  # system to determine when new messages the user has not seen have been
-  # added to the motd.
-  def motd_config_file
-    @motd_config_file ||= OodAppkit.dataroot.join(".motd")
+  # @param [String] path The path to the motd file as a URI
+  def initialize(path = ENV['MOTD_PATH'])
+    @motd_uri = path
   end
 
+  # Checks the path URI to see if it can be opened
+  #
+  # Uses open-uri to check local or remote path for contents
   def exist?
-    motd_system_file && File.file?(motd_system_file)
+    exists = false
+    if motd_uri
+      begin
+        open(motd_uri)
+        exists = true
+      rescue
+        # The messages file does not exist.
+        Rails.logger.warn "MOTD File is missing; it was expected at #{motd_uri}"
```

Comment:
I don't know how I feel about this logging at all, actually. I worry that if a site doesn't set up a `motd` we're going to be bombarding their logs with error messages about motd files not present every time a user hits the dashboard.

It may be better to catch and move on from expected IO, HTTP, and FTP errors and only log errors that don't match, or when `motd_uri` isn't blank.